tests/github_client/mod.rs

```diff
@@ -0,0 +1,876 @@
+//! `GithubClient` tests.
+//!
+//! These tests exercise the behavior of `GithubClient`. They involve setting
+//! up HTTP servers, creating a `GithubClient` to connect to those servers,
+//! executing some action, and validating the result.
+//!
+//! The [`run_test`] function is used to set up the test and give you a
+//! [`GithubClient`] to perform some action and validate its result.
+//!
+//! To write one of these tests, you'll need to use the recording function
+//! against the live GitHub site to fetch what the actual JSON objects should
+//! look like. To write a test, follow these steps:
+//!
+//! 1. Choose a repo in your GitHub account to run tests against. I use either
+//!    `ehuss/rust` or `ehuss/triagebot-test`, but any in your account will do.
+//!
+//!    **WARNING**: Do not write tests that modify the rust-lang org repos
+//!    like rust-lang/rust. Write those tests against your own fork (like
+//!    `ehuss/rust`). We don't want to pollute the real repos with things like
+//!    test PRs.
```

Comment:
I don't quite understand this sentence (but mybe it's just me)

Reply:
I'll add some clarity to that. I was just meaning, we don't want tests making changes or opening PRs on production repos like https://github.com/rust-lang/rust/.